### PR TITLE
ISPN-11148 + ISPN-11149 + ISPN-11150 CLI hang fix / CLI logging

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-11148 CLI: close all connections to the server to prevent hang
https://issues.redhat.com/browse/ISPN-11149 CLI logging file
https://issues.redhat.com/browse/ISPN-11150 Show server URL on boot

### CODE REVIEW

File: server/runtime/src/main/java/org/infinispan/server/logging/Log.java [hunk 1]
```diff
@@ -132,4 +132,8 @@
 
    @Message(value = "Cannot find a network address which matches the supplied configuration", id = 80033)
    CacheConfigurationException invalidNetworkConfiguration();
+
+   @LogMessage(level = Logger.Level.INFO)
+   @Message(value = "Server listening on %s://%s:%d", id = 80034)
```
Comment (danberindei) [inline server/runtime/src/main/java/org/infinispan/server/logging/Log.java]: Do we have a server name we could add to the log?

Comment (tristantarrant) [reply server/runtime/src/main/java/org/infinispan/server/logging/Log.java]: We could use the node name. Is that what you mean ?

`Server 'fluffy' listening on https://localhost:11222`

File: cli/cli-client/src/main/java/org/infinispan/cli/impl/ContextImpl.java [hunk 2]
```diff
@@ -80,6 +81,7 @@ public Connection connect(Shell shell, String connectionString) {
          connection.connect();
       } catch (AccessDeniedException accessDenied) {
          try {
+            connection.close();
```
[the remark this answers is not in the document]
Comment (tristantarrant) [reply cli/cli-client/src/main/java/org/infinispan/cli/impl/ContextImpl.java]: Sure

File: server/runtime/src/main/java/org/infinispan/server/logging/Log.java (hunk 1 above)
Comment (tristantarrant) [reply server/runtime/src/main/java/org/infinispan/server/logging/Log.java]: Done

**Comment by tristantarrant** [comment]
@danberindei repushed

File: server/runtime/src/main/java/org/infinispan/server/logging/Log.java (hunk 1 above)
Comment (danberindei) [reply server/runtime/src/main/java/org/infinispan/server/logging/Log.java]: Yeah, looks good!

**Comment by danberindei** [comment]
Integrated, thanks Tristan!

**Pull request merged**